npm_and_yarn/lib/dependabot/npm_and_yarn/package_name.rb

```diff
@@ -19,7 +19,7 @@ class PackageName
             [a-z0-9\-\_\.\!\~\*\'\(\)]+               # URL-safe characters
           )\/)?
           (?<name>                                    # capture package name
-            (?=[^\.\_])                               # reject leading dot or underscore
+            (?=[^\.])                               # reject leading dot or underscore
```

Comment:
can we verify that leading dots are also not acceptable in npm going forward? If so, we should update the comment with a link to that. If not, we should fix both issues in one go.